File: swag-labs/playwright-cs/Pages/LoginPage.cs
using Microsoft.Playwright;

namespace Pages;

// Page Object da tela de login do Swag Labs.
// Todos os seletores e ações dessa tela ficam centralizados aqui.
public class LoginPage : BasePage
{
    private const string UsernameSelector = "#user-name";
    private const string PasswordSelector = "#password";
    private const string SubmitSelector = "#login-button";
    private const string ErrorSelector = "[data-test='error']";

    public LoginPage(IPage page) : base(page) { }

    public async Task OpenAsync()
    {
        await NavigateToAsync("/");
        await Page.Locator(UsernameSelector).WaitForAsync(new LocatorWaitForOptions { State = WaitForSelectorState.Visible });
    }

    public async Task LoginAsync(string username, string password)
    {
        await Page.Locator(UsernameSelector).FillAsync(username);
        await Page.Locator(PasswordSelector).FillAsync(password);
        await Page.Locator(SubmitSelector).ClickAsync();
    }

    public async Task<string> GetErrorMessageAsync()
    {
        return await Page.Locator(ErrorSelector).InnerTextAsync();
    }
}

File: swag-labs/playwright-cucumber-ts/src/pages/loginPage.ts
import type { Page } from '@playwright/test';

// Page Object da tela de login do Swag Labs.
export class LoginPage {
  constructor(private readonly page: Page) {}

  private usernameSelector = '#user-name';
  private passwordSelector = '#password';
  private submitSelector = '#login-button';
  private errorSelector = '[data-test="error"]';

  async open() {
    await this.page.goto(process.env.BASE_URL || 'https://www.saucedemo.com');
  }

  async login(username: string, password: string) {
    await this.page.fill(this.usernameSelector, username);
    await this.page.fill(this.passwordSelector, password);
    await this.page.click(this.submitSelector);
  }

  async getErrorMessage() {
    return this.page.locator(this.errorSelector).innerText();
  }
}

